src/main/java/eu/dissco/exportjob/repository/ElasticSearchRepository.java

```diff
@@ -0,0 +1,99 @@
+package eu.dissco.exportjob.repository;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import co.elastic.clients.elasticsearch.core.CountRequest;
+import co.elastic.clients.elasticsearch.core.SearchRequest;
+import co.elastic.clients.elasticsearch.core.search.Hit;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import eu.dissco.exportjob.domain.SearchParam;
+import eu.dissco.exportjob.domain.TargetType;
+import eu.dissco.exportjob.properties.ElasticSearchProperties;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class ElasticSearchRepository {
+  private final ElasticsearchClient client;
+  private final ElasticSearchProperties properties;
+
+  public Long getTotalHits(List<SearchParam> searchParams, TargetType targetType)
+      throws IOException {
+    var query = generateQuery(searchParams);
+    var index = getIndex(targetType);
+    var countRequest = new CountRequest.Builder()
+        .index(index)
+        .query(
+            q -> q.bool(b -> b.must(query)))
+        .build();
+    return client.count(countRequest).count();
+  }
+
+  public List<JsonNode> getTargetObjects(List<SearchParam> searchParams, TargetType targetType,
+      int pageNumber)
+      throws IOException {
+    var query = generateQuery(searchParams);
+    var index = getIndex(targetType);
+
+    var searchRequest = new SearchRequest.Builder()
+        .index(index)
+        .query(
+            q -> q.bool(b -> b.must(query)))
+        .trackTotalHits(t -> t.enabled(Boolean.TRUE))
+        .from(getOffset(pageNumber, properties.getPageSize()))
+        .size(properties.getPageSize())
+        .build();
+    var searchResult = client.search(searchRequest, ObjectNode.class);
```

Comment:
Maybe you could only return the selected fields from elastic with https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html